middleware/async.js

```diff
@@ -0,0 +1,5 @@
+const asyncHandler = (fn) => (req, res, next) => {
+  Promise.resolve(fn(req, res, next)).catch(next);
+};
+
+module.exports = asyncHandler;
```

Comment:
purpose of using async Handlers as middleware ?

Reply:
I'm applying the dry principle here. since I'm using the async function, I'm going to repeat try/catch. asyncHandler prevents them from repeating. It helps to make code clean and readable.